### PR TITLE
Change http client for batch subscriptions

### PR DESCRIPTION
**Pull request by moscicky**
This PR resolves: https://github.com/allegro/hermes/issues/1619, https://github.com/allegro/hermes/issues/1011

There is a separate jetty http client for batch subscriptions, configurable via:

`consumer.batch.httpClient` - properties the same as for regular serial client. 

I have cherry picked 579226ce95bd0dacb9d574399b3231ca279701a9 to master and can confirm that the test included in that commit doesn't pass there (because apache supports only socket timeout)

### CODE REVIEW

**Comment by piotrrzysko** [comment]
@moscicky Am I right that this will also solve #1011?

**Comment by moscicky** [comment]
> @moscicky Am I right that this will also solve #1011?

@piotrrzysko You are right, added the issue to the PR description

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/config/ConsumerConfiguration.java [hunk 1]
```diff
@@ -161,6 +161,11 @@ public HttpClient http1Client(HttpClientsFactory httpClientsFactory) {
         return httpClientsFactory.createClientForHttp1("jetty-http-client");
```
Comment (faderskd) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/config/ConsumerConfiguration.java:161]: Maybe we should change the previous client name to the `jetty-http-serial-client` to stay consistent ?

**Comment by piotrrzysko** [comment]
It would be good to have the metrics reported by `HttpClientsWorkloadReporter` also for batches. These metrics are important as both the connection pool and requests queue are configured statically, so we need data to tune them correctly.

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/DefaultSendingResultHandlers.java [hunk 2]
```diff
@@ -19,7 +19,7 @@ public Response.CompleteListener handleSendingResultForBroadcast(CompletableFutu
     }
 
     @Override
-    public MessageSendingResult handleSendingResultForBatch(CloseableHttpResponse response) {
-        return MessageSendingResult.ofStatusCode(response.getStatusLine().getStatusCode());
+    public MessageSendingResult handleSendingResultForBatch(ContentResponse response) {
```
Comment (piotrrzysko) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/DefaultSendingResultHandlers.java:22]: I wonder if it is feasible to make this API simpler - ideally, I would like to have one handler for all types of subscriptions. 

This is not strictly related to this PR. I'm just curious if we can somehow leverage the fact that now we use the same HTTP client everywhere. Plus, I'm not sure if we have the capacity for a bigger refactoring. Let's discuss this next week.

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/HttpClientsFactory.java [hunk 3]
```diff
@@ -26,6 +26,10 @@ public HttpClientsFactory(HttpClientParameters httpClientParameters,
     }
 
     public HttpClient createClientForHttp1(String name) {
```
[the remark this answers is not in the document]
Comment (moscicky) [reply hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/HttpClientsFactory.java]: https://github.com/allegro/hermes/pull/1620/commits/f6775979acc592a175f4f68a6ca2c1cd74112520 added batch client to workload metrics and refactored the configuration so it now explicit which client is configured by what properties:

For example:

```yaml
  http-client:
    monitoring:
      connectionPoolMonitoringEnabled: false
      requestQueueMonitoringEnabled: true
    serial:
      http1:
        threadPoolSize: 30
        threadPoolMonitoringEnabled: false
        followRedirectsEnabled: false
        maxConnectionsPerDestination: 100
        idleTimeout: 0ms
        maxRequestsQueuedPerDestination: 100
      http2:
        enabled: true
        threadPoolSize: 10
        threadPoolMonitoringEnabled: false
        idleTimeout: 0ms
        followRedirectsEnabled: false
        maxRequestsQueuedPerDestination: 100
    batch:
      http1:
        threadPoolSize: 30
        threadPoolMonitoringEnabled: false
        followRedirectsEnabled: false
        maxConnectionsPerDestination: 100
        idleTimeout: 0ms
        maxRequestsQueuedPerDestination: 100
```

oauth client is still using `http-client.serial.http1` but can be configured separately if its needed in the future

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/config/ConsumerConfiguration.java (hunk 1 above)
Comment (moscicky) [reply hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/config/ConsumerConfiguration.java]: fixed

**Review by piotrrzysko — DISMISSED** [review]

**Review by piotrrzysko — APPROVED** [review]

**Review by piotrrzysko — APPROVED** [review]

**Review by faderskd — APPROVED** [review]
